tensorflow/contrib/avro/ops/parse_avro_record.cc

```diff
@@ -0,0 +1,2058 @@
+// the attributes have the correct values.
+//
+// returns OK if all attributes are valid; otherwise false.
+Status ParseAvroAttrs::FinishInit() {
+  if (static_cast<size_t>(num_sparse) != sparse_types.size()) {
+    return errors::InvalidArgument("len(sparse_keys) != len(sparse_types)");
+  }
+  if (static_cast<size_t>(num_dense) != dense_infos.size()) {
+    return errors::InvalidArgument("len(dense_keys) != len(dense_infos)");
+  }
+  if (num_dense > std::numeric_limits<int32>::max()) {
+    return errors::InvalidArgument("num_dense_ too large");
+  }
+  for (const DenseInformation& dense_info : dense_infos) {
+    TF_RETURN_IF_ERROR(CheckValidType(dense_info.type));
+  }
+  for (const DataType& type : sparse_types) {
+    TF_RETURN_IF_ERROR(CheckValidType(type));
+  }
+  return Status::OK();
+}
+
+// Register the parse function when building the shared library
+// For the op I used as boiler plate: tensorflow/core/ops/parsing_ops.cc and
+// there 'ParseExample'
+// For the op kernel I used as boiler plate:
+// tensorflow/core/kernels/example_parsing_ops.cc and there 'ExampleParserOp'
+// For the compute method I used as boiler plate:
+// tensorflow/core/util/example_proto_fast_parsing.cc and there the
+//   method 'FastParseExample'
+
+REGISTER_OP("ParseAvroRecord")
+    .Input("serialized: string")
+    .Input("sparse_keys: Nsparse * string")
+    .Input("dense_keys: Ndense * string")
+    .Input("dense_defaults: Tdense")
+    .Output("sparse_indices: Nsparse * int64")
+    .Output("sparse_values: sparse_types")
+    .Output("sparse_shapes: Nsparse * int64")
+    .Output("dense_values: Tdense")
+    .Attr("Nsparse: int >= 0")  // Inferred from sparse_keys
+    .Attr("Ndense: int >= 0")   // Inferred from dense_keys
+    .Attr("sparse_types: list({float,double,int64,int32,string,bool}) >= 0")
+    .Attr("Tdense: list({float,double,int64,int32,string,bool}) >= 0")
+    .Attr("dense_shapes: list(shape) >= 0")
+    .Attr("schema: string")
+    .SetShapeFn([](::tensorflow::shape_inference::InferenceContext* c) {
+
+       ParseAvroAttrs attrs;
+       TF_RETURN_IF_ERROR(attrs.Init(c));
+
+       // Get the batch size and load it into input
+       ShapeHandle input;
+       TF_RETURN_IF_ERROR(c->WithRank(c->input(0), 1, &input));
+
+       // Get the schema, parse it, and log it
+       string schema;
+       TF_RETURN_IF_ERROR(c->GetAttr("schema", &schema));
+
+       std::unique_ptr<avro_schema_t, std::function<void(avro_schema_t*)>>
+           p_reader_schema(new avro_schema_t, [](avro_schema_t* ptr) {
+             avro_schema_decref(*ptr);
+           });
+       if (avro_schema_from_json_length(schema.c_str(), schema.length(),
+                                        &(*p_reader_schema)) != 0) {
+         return errors::InvalidArgument("The provided json schema is invalid. ",
+                                        avro_strerror());
+       }
+       LOG(INFO) << "Avro parser with schema\n" << schema;
```

Comment:
This gets output for literally every avro record that gets parsed. I think it would be good to set the log level to something like DEBUG instead.